wremnants/syst_tools.py

```diff
@@ -284,23 +285,56 @@ def add_QCDbkg_jetPt_hist(results, df, nominal_axes, nominal_cols, base_name="no
     dQCDbkGVar = dQCDbkGVar.Filter(f"passMT || Sum(goodCleanJetsPt{jet_pt})>=1")
     qcdJetPt = dQCDbkGVar.HistoBoost(name, nominal_axes, [*nominal_cols, "nominal_weight"], storage=hist.storage.Double())
     results.append(qcdJetPt)
-                                        
-def add_muon_efficiency_unc_hists(results, df, helper_stat, helper_syst, axes, cols, base_name="nominal", is_w_like=False):
 
-    if is_w_like:
-        muon_columns_stat = ["trigMuons_pt0", "trigMuons_eta0", "trigMuons_charge0", "nonTrigMuons_pt0", "nonTrigMuons_eta0", "nonTrigMuons_charge0"]
-        muon_columns_syst = ["trigMuons_pt0", "trigMuons_eta0", "trigMuons_SApt0", "trigMuons_SAeta0", "trigMuons_charge0",
-            "nonTrigMuons_pt0", "nonTrigMuons_eta0", "nonTrigMuons_SApt0", "nonTrigMuons_SAeta0", "nonTrigMuons_charge0"]
+def add_muon_efficiency_unc_hists(results, df, helper_stat, helper_syst, axes, cols, base_name="nominal", what_analysis=ROOT.wrem.AnalysisType.Wmass, smooth3D=False):
+    # TODO: update for dilepton
+    if what_analysis == ROOT.wrem.AnalysisType.Dilepton:
```

Comment:
Can try, but here it is really the fact that the helpers require different input variables, although usually the sorting has been kept the same (i.e. pt, eta, ut (if present), charge, passOtherThings, ...).
The Wlike arrays are not simply Wmass with added variables, and similarly Dilepton is not necessarily just adding variables on top of Wlike

Reply:
Ok, but the point stands that it could be more clear what's different between the two when it's a list of 10+ things. You can also use a dictionary in python like

args = {arg1=blah1, arg2=blah2...}
func_call(**args)

Then you just update the args dict for the additional wlike/dilepton args.That could be a little clearer.